FileNotFoundException.h

```diff
@@ -0,0 +1,18 @@
+#ifndef FILENOTFOUNDEXCEPTIION_H
+#define FILENOTFOUNDEXCEPTIION_H
+
+#include <iostream>
+
+class FileNotFoundException : public std::exception
+{
+public:
+	FileNotFoundException(std::string message) :msg(message) {}
+	~FileNotFoundException() {}
+
+	std::string message() { return msg; }
+
+private:
+	std::string msg;
+};
+
+#endif
```

Comment:
Alapvetoen ez igy rendben van, de igy semmi ertelme nincs az std::exception-bol valo szarmaztatasnak. Ha abbol szarmaztattok, akkor a `what`-ot illene feluldefinialni. Es mivel a message az lesz, hogy "File not found", ezert a konstans literalert teljesen felesleges felvenni egy string adattagot. Arra akkor lenne szukseg, ha a fajl nevet is kiirnatok, ami amugy nem lenne hulyeseg. Es persze akkor ez const, mert nem nagyon valtozik.

Vagy mivel az exception-on kivul siman csak egy string-et tarol az egesz... akar szarmazhatna abbol is, akkor rogton meglenne ra a << operator meg minden. Magic.